### PR TITLE
[SIW-1857] Disable functions in staging slot

### PR DESCRIPTION
**Pull request by silvicir**
#### List of Changes

Added variables with deployment slot setting enabled

#### Motivation and Context

The non-HTTP-triggered functions need to be disabled in the staging slot

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: infra/resources/_modules/function_apps/locals.tf [hunk 1]
```diff
@@ -1,50 +1,69 @@
...
+    WEBSITE_SWAP_WARMUP_PING_STATUSES = "200"
+    },
+    local.function_apps.common_app_settings,
+    {
+      for s in var.user_func.app_settings :
+      s.name => s.key_vault_secret_name != null ? "@Microsoft.KeyVault(VaultName=${var.key_vault_wallet_name};SecretName=${s.key_vault_secret_name})" : s.value
+    }
+  )
+
+  function_app_user_staging_disabled = [
```
Comment (Krusty93) [inline infra/resources/_modules/function_apps/locals.tf:50]: I think you should put the entire variable name like `AzureWebJobs.addWalletInstanceToValidationQueue.Disabled`

File: infra/resources/_modules/function_apps/locals.tf [hunk 2]
```diff
@@ -1,50 +1,69 @@
...
+    "generateEntityConfiguration"
+  ]
+
+  function_app_user = {
+    app_settings = merge(local.function_app_user_common_app_settings,
       {
-        for s in var.user_func.app_settings :
-        s.name => s.key_vault_secret_name != null ? "@Microsoft.KeyVault(VaultName=${var.key_vault_wallet_name};SecretName=${s.key_vault_secret_name})" : s.value
+        for to_disable in local.function_app_user_staging_disabled :
+        format("AzureWebJobs.%s.Disabled", to_disable) => false
```
Comment (Krusty93) [inline infra/resources/_modules/function_apps/locals.tf:60]: the value shall be `0` or `1`

File: infra/resources/_modules/function_apps/locals.tf [hunk 3]
```diff
@@ -1,50 +1,69 @@
...
+      }
+    )
+    slot_app_settings = merge(local.function_app_user_common_app_settings,
+      {
+        for to_disable in local.function_app_user_staging_disabled :
+        format("AzureWebJobs.%s.Disabled", to_disable) => true
```
Comment (Krusty93) [inline infra/resources/_modules/function_apps/locals.tf:66]: ditto

File: infra/resources/_modules/function_apps/locals.tf (hunk 1 above)
Comment (silvicir) [reply infra/resources/_modules/function_apps/locals.tf:50]: below I put `format("AzureWebJobs.%s.Disabled", to_disable) => false`

File: infra/resources/_modules/function_apps/locals.tf [hunk 4]
```diff
@@ -1,50 +1,69 @@
...
+    "generateEntityConfiguration"
+  ]
+
+  function_app_user = {
+    app_settings = merge(local.function_app_user_common_app_settings,
       {
-        for s in var.user_func.app_settings :
-        s.name => s.key_vault_secret_name != null ? "@Microsoft.KeyVault(VaultName=${var.key_vault_wallet_name};SecretName=${s.key_vault_secret_name})" : s.value
+        for to_disable in local.function_app_user_slot_disabled :
+        format("AzureWebJobs.%s.Disabled", to_disable) => 0
```
Comment (mamu0) [inline infra/resources/_modules/function_apps/locals.tf:60]: It is unclear why `to_disable` here is being set to 0 and why for the slot it is set to 1

**Review by mamu0 — APPROVED** [review]

**Pull request merged**